drive/src/query/mod.rs

```diff
@@ -1269,7 +1269,26 @@ impl<'a> DriveQuery<'a> {
                 let document_holding_path = self
                     .contract
                     .documents_primary_key_path(self.document_type.name.as_str());
-                let start_at_document = grove.get(document_holding_path, starts_at, transaction)?;
+
+                let start_at_document =
+                    match grove.get(document_holding_path, starts_at, transaction) {
+                        Ok(element) => element,
+                        Err(error) => {
+                            return match error {
+                                Error::PathKeyNotFound(_) => {
+                                    let error_message = if self.start_at_included {
+                                        "startAt document not found"
+                                    } else {
+                                        "startAfter document not found"
+                                    };
+
+                                    Err(Error::InvalidQuery(error_message))
+                                }
+                                _ => Err(error),
+                            }
+                        }
+                    };
+
```

Comment:
you can use `.map_err` to get rid of outer `match` with following `?` sign for early return on error

Reply:
Good point!